portals/devportal/source/src/app/components/Apis/Details/SolaceApi/SolaceTopicsInfo.jsx

```diff
@@ -0,0 +1,580 @@
+    }
+    // Handle topic changes with protocol change
+    function setProtocolTopics(protocol, selectedEnv) {
+        if (protocol === 'mqtt') {
+            setTopics(selectedEnv.SolaceTopicsObject.mqttSyntax);
+            setPubAndSubTopics(selectedEnv.SolaceTopicsObject.mqttSyntax, apiTopics);
+        } else {
+            setTopics(selectedEnv.SolaceTopicsObject.defaultSyntax);
+            setPubAndSubTopics(selectedEnv.SolaceTopicsObject.defaultSyntax, apiTopics);
+        }
+    }
+    // Handle application selection change
+    const handleChangeApplication = (event) => {
+        setApplication(event.target.value);
+        let selectedApp;
+        applicationList.map((app) => {
+            if (app.applicationName === event.target.value) {
+                selectedApp = app;
+            }
+            return null;
+        });
+        const selectedEnv = selectedApp.solaceDeployedEnvironments[0];
+        setEnvsListOfApplication(selectedApp.solaceDeployedEnvironments);
+        setSelectedProtocol(selectedEnv.solaceURLs[0].protocol);
+        let protocol;
+        selectedEnv.solaceURLs.map((e) => {
+            if (e.protocol === selectedEnv.solaceURLs[0].protocol) {
+                setSelectedEndpoint(e.endpointURL);
+                protocol = e.protocol;
+            }
+            return null;
+        });
+        setProtocolTopics(protocol, selectedEnv);
+    };
+    // Handle environment selection change
+    const handleChangeEnvironment = (event) => {
+        setEnvironment(event.target.value);
+        let selectedEnv;
+        envsListOfApplication.map((e) => {
+            if (e.environmentDisplayName === event.target.value) {
+                selectedEnv = e;
+            }
+            return null;
+        });
+        setSelectedProtocol(selectedEnv.solaceURLs[0].protocol);
+        setProtocolsListOfEnvironments(selectedEnv.solaceURLs);
+        let protocol;
+        selectedEnv.solaceURLs.map((e) => {
+            if (e.protocol === selectedEnv.solaceURLs[0].protocol) {
+                setSelectedEndpoint(e.endpointURL);
+                protocol = e.protocol;
+            }
+            return null;
+        });
+        setProtocolTopics(protocol, selectedEnv);
+    };
+    // Handle protocol selection change
+    const handleChangeProtocol = (event) => {
+        setSelectedProtocol(event.target.value);
+        let selectedEnv;
+        envsListOfApplication.map((e) => {
+            if (e.environmentDisplayName === environment.environmentDisplayName) {
+                selectedEnv = e;
+            }
+            return null;
+        });
+        let protocol;
+        protocolsListOfEnvironments.map((e) => {
+            if (e.protocol === event.target.value) {
+                setSelectedEndpoint(e.endpointURL);
+                protocol = e.protocol;
+            }
+            return null;
+        });
+        setProtocolTopics(protocol, selectedEnv);
+    };
+
+
+    console.log(apiTopics);
```

Comment:
Better not to leave the logs

Reply:
Fixed with https://github.com/wso2/apim-apps/pull/81/commits/70c7422e78f3f2ac41db0abbab5b8ebcc8faa27c